src/s2e/Plugins/ExecutionTracers/TestCaseGenerator.cpp

```diff
@@ -303,15 +315,77 @@ bool TestCaseGenerator::assembleChunks(const TestCaseFile &file, std::vector<uin
 /// v0___symfile____tmp_input___0_1_symfile___0 = {0x0, .... }
 /// This is the first (0) chunk of a file that has only one (1) chunk. That file was called /tmp/input.
 ///
+/// \param file the chunked representation of the concrete file
+/// \param out the assembled file content
+/// \return true if assembling was successful
 ///
-/// \param inputs the concrete inputs that must contain symbolic file chunks
-/// \param prefix the prefix to add to the generated test case file
-/// \param fileNames returns the location of the written file names
-///
-void TestCaseGenerator::assembleTestCaseToFiles(const ConcreteInputs &inputs, const std::string &prefix,
-                                                std::vector<std::string> &fileNames) {
+bool TestCaseGenerator::assembleChunks(const std::string &name, const TestCaseFile &file,
+                                       const ConcreteFileTemplates &templates, std::vector<uint8_t> &out) {
+    const auto tpl = templates.lookup_previous(name);
+
+    if (tpl == nullptr) {
+        // The file has no concrete template, so all its data must be contained
+        // in symbolic variables.
+        if (file.totalParts != file.chunks.size()) {
+            getWarningsStream() << "Test case has incorrect number of parts\n";
+            return false;
+        }
+
+        if (file.chunksData.size() != file.chunks.size()) {
+            getWarningsStream() << "Test case has not enough data chunks\n";
+            return false;
+        }
+
+        // Get the total size
+        // The loop supposes the chunks are traversed in increasing order
+        unsigned size = 0;
+        std::vector<unsigned> offsets;
+        foreach2 (it, file.chunks.begin(), file.chunks.end()) {
```

Comment:
Did you try a c++11 loop here and in the subsequent loops?

Reply:
I guess I just copied to old code without refactoring it, I can fix it.